packages/backend-test-utils/src/next/services/mockServices.ts

```diff
@@ -57,6 +57,17 @@ import {
   eventsServiceRef,
 } from '@backstage/plugin-events-node';
 
+/** @internal */
+function createLoggerMock() {
+  return {
+    child: jest.fn().mockImplementation(createLoggerMock),
```

Comment:
> [!Note]
> For some tests, this was failing because the child function returned `undefined` previously. There is a fragility with this solution, if the `resetAllMocks` is used, the implementation mock will be removed and the tests calling `child` will fail again. Any suggestions to prevent this from happening?

Reply:
Hmm, if you use `jest.fn(createLoggerMock)` here instead, does that have an effect?